### PR TITLE
[Tracker Alignment] remove `getByLabel` and allow to fetch vertex collection from configuration

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Trivial bug-fix in debugging modules. No regression expected in any tests.

#### PR validation:

Private, but it will be tested by unit test.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by yuanchao** [comment]
@cmsbuild please test

**Comment by mmusich** [comment]
@cmsbuild, please abort.
IBs are not yet clean, no point in testing now

File: Alignment/OfflineValidation/plugins/DMRChecker.cc [hunk 1]
```diff
@@ -1036,8 +1036,7 @@ class DMRChecker : public edm::one::EDAnalyzer<edm::one::SharedResources> {
       //dxy with respect to the primary vertex
       reco::Vertex pvtx;
       edm::Handle<reco::VertexCollection> vertexHandle;
-      reco::VertexCollection vertexCollection;
-      event.getByLabel("offlinePrimaryVertices", vertexHandle);
+      event.getByToken(vertexToken, vertexHandle);
```
Comment (Dr15Jones) [inline Alignment/OfflineValidation/plugins/DMRChecker.cc:1039]: Thanks for the change! FYI an even more compact and safe version would be 
```C++
      edm::Handle<reco::VertexCollection> vertexHandle = event.getHandle(vertexToken);
```

Comment (tvami) [reply Alignment/OfflineValidation/plugins/DMRChecker.cc:1039]: Hi @mmusich do you plan to do this change?

Comment (mmusich) [reply Alignment/OfflineValidation/plugins/DMRChecker.cc:1039]: not now.

Comment (mmusich) [reply Alignment/OfflineValidation/plugins/DMRChecker.cc:1039]: @tvami please clarify if the change is necessary for signature. As the code is now, it is buggy (a part from not compliant with the deprecation rules).

Comment (tvami) [reply Alignment/OfflineValidation/plugins/DMRChecker.cc:1039]: It's not necessary, I just didnt understand if "not now" means 1-2 hours later, or in a next PR. From this msg I understand it means not in this PR. I'll sign now.

**Comment by tvami** [comment]
+alca
 * tests pass
 * possible (compactifying) improvement discussed is for a future update

**Comment by perrotta** [comment]
+1

File: Alignment/OfflineValidation/plugins/DMRChecker.cc (hunk 1 above)
Comment (mmusich) [reply Alignment/OfflineValidation/plugins/DMRChecker.cc:1039]: addressed at https://github.com/cms-sw/cmssw/commit/309aa0ed5edd8c7502f65b3d8ec4e2743117ecf9

**Pull request merged**